### PR TITLE
docs: update Logging section in `README` and `CONTRIBUTING.md`

### LINKED ISSUE #843: Update `logging` section in readme
Version 0.8.2 uses a new logging based on `debug`, but the [readme still documents the old way of logging](https://github.com/eclipse/thingweb.node-wot/#logging).

### PR DESCRIPTION
**Pull request by JKRhb**
As raised in #843, this PR updates the README file with information regarding the new logging approach and provides examples for both users and package implementors.

Resolves #843.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -402,14 +402,69 @@ To learn by examples, see `examples/scripts` to have a feeling of how to script
...
-We used to have a node-wot-logger package to allow fine-grained logging (by means of Winston). It turned out though that depending on the actual use-case other logging libraries might be better suited. Hence we do not want to prescribe which logging library to use. Having said that, we use console statements which can be easily overriden to use the prefered logging library if needed (see [here](https://gist.github.com/spmason/1670196)).
+Logging in node-wot is implemented via the [`debug`](https://www.npmjs.com/package/debug) package.
+This allows users to enable log messages for specific logging levels (`info`, `debug`, `warn`, or `error`) or packages.
+Which log messages are emitted is controlled by the `DEBUG` environment variable.
 
-The logs in the library follows those best practice rules (see [here](https://github.com/eclipse/thingweb.node-wot/issues/229)):
+In the following, we will show a couple of examples for its usage using wildcard characters (`*`).
+Note, however, that the syntax for setting an environment variable depends on your operating system and the terminal you use.
+See the `debug` documentation linked above for more details on platform-specific differences.
```
Comment (relu91) [inline README.md:411]:
```suggestion
See the [`debug` documentation](https://www.npmjs.com/package/debug) for more details on platform-specific differences.
```

File: README.md [hunk 2]
```diff
@@ -402,14 +402,69 @@ To learn by examples, see `examples/scripts` to have a feeling of how to script
...
+import { createLoggers } from "@node-wot/core";
+const { debug, info, warn, error } = createLoggers("binding-foo", "foo-server");
+
+function startFoo() {
+    info("This is an info message!");
+    debug("This is a debug message!");
+    warn("This is a warn message!");
+    error("This is an error message!");
+}
+```
```
Comment (relu91) [inline README.md:467]: This content should be inside the Contributing.md file and here we should only give a referenced to the relevant section. We can move the section later.

**Review by relu91 — CHANGES_REQUESTED** [review]
Thank you @JKRhb for your great contributions 😃 . I would change just two bits below.

File: README.md (hunk 2 above)
Comment (JKRhb) [reply README.md:467]: Oh, yeah, that is a good idea! :)

Comment (JKRhb) [reply README.md:467]: Hmm, I suppose we could add a new section like "Adding a new Protocol Binding" to the `Contributing.md` for this purpose?

Comment (relu91) [reply README.md:467]: > Hmm, I suppose we could add this to a new section like "Adding a new Protocol Binding", containing also more general instructions for implementors?

Yes, good point!

File: README.md [hunk 3]
```diff
@@ -402,14 +402,69 @@ To learn by examples, see `examples/scripts` to have a feeling of how to script
...
+
+```sh
+DEBUG=node-wot:core* npm start
+```
+
+Using the log levels above, you can also apply more fine-grained parameters for logging.
+For instance, if you only want to see `error` messages from the `binding-coap` package, use this:
+
+```sh
+DEBUG=node-wot:binding-coap*debug npm start
```
Comment (danielpeintner) [inline README.md:446]: I think it should be 

DEBUG=node-wot:binding-coap*error npm start

File: README.md (hunk 2 above)
Comment (danielpeintner) [reply README.md:467]: Makes sense to move.

File: README.md (hunk 3 above)
Comment (JKRhb) [reply README.md]: Thank you, fixed in 73a8da2f206ca232323ff5ca709ffe622ab358a0 :)

**Comment by JKRhb** [comment]
> @JKRhb do you want to address the comments above in a future PR?

Oh, yeah, I think that sounds good :) I would create a new PR right after this one has been merged.

**Comment by JKRhb** [comment]
Hmm, yeah, you're right. I'll update the PR shortly.

**Comment by JKRhb** [comment]
I now started a new section on adding protocol bindings in the `CONTRIBUTING.md` file. There is probably still room for improvement, but I guess it is a start. Feel free to make suggestions!

**Review by danielpeintner — APPROVED** [review]

**Review by relu91 — APPROVED** [review]

**Pull request merged**